x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -256,6 +261,46 @@ private static void checkInsist(LogicalPlan p, Failures failures) {
         }
     }
 
+    /*
+     * This is a rudimentary check to prevent INLINE STATS after LIMIT. A LIMIT command can be added by other commands by default,
+     * the best example being FORK. A more robust solution would be to track the commands that add LIMIT and prevent them from doing so
+     * if INLINE STATS is present in the plan. However, this would require authors of new such commands to be aware of this limitation and
+     * implement the necessary checks, which is error-prone.
+     */
+    private static void checkLimitBeforeInlineStats(LogicalPlan plan, Failures failures) {
+        if (plan instanceof InlineStats is) {
+            Holder<Limit> inlineStatsDescendantLimit = new Holder<>();
+            is.forEachDownMayReturnEarly((p, breakEarly) -> {
+                if (p instanceof Limit l) {
+                    inlineStatsDescendantLimit.set(l);
+                    breakEarly.set(true);
+                    return;
+                }
+            });
+
+            var firstLimit = inlineStatsDescendantLimit.get();
+            if (firstLimit != null) {
+                Location l = firstLimit.source().source();
+                var limitLocation = "line " + l.getLineNumber() + ":" + l.getColumnNumber();
+                var isString = is.sourceText().length() > Node.TO_STRING_MAX_WIDTH
+                    ? is.sourceText().substring(0, Node.TO_STRING_MAX_WIDTH) + "..."
+                    : is.sourceText();
+                var limitString = firstLimit.sourceText().length() > Node.TO_STRING_MAX_WIDTH
+                    ? firstLimit.sourceText().substring(0, Node.TO_STRING_MAX_WIDTH) + "..."
+                    : firstLimit.sourceText();
+                failures.add(
+                    fail(
+                        is,
+                        "INLINE STATS cannot be used after LIMIT command, but was [{}] after [{}] at [{}]",
```

Comment:
```suggestion
                        "INLINE STATS cannot be used after an explicit or implicit LIMIT command, but was [{}] after [{}] at [{}]",
```
I'd add this clarification since with FORK it's not otherwise at all clear why the query's rejected. We do document that FORK adds a default LIMIT.